csm_web/scheduler/views.py

```diff
@@ -73,21 +76,31 @@ def students(self, request, pk=None):
             the section directly, any student trying to enroll must first acquire a lock on the
             desired section. This allows us to assume that current_student_count is correct.
             """
+            section = get_object_or_error(Section.objects, pk=pk)
             section = Section.objects.select_for_update().get(pk=section.pk)
             if not request.user.can_enroll_in_course(section.course):
+                logger.warn(
+                    f"<Enrollment:Failure> User {request.user.pk} was unable to enroll in Section {section.pk} because they are already involved in this course")
```

Comment:
Would be useful to display the __str__ as well in case the section mutates

Reply:
(Also makes much more readable, esp if we're searching the logs for users - saves us the time of going to django shell to check)

Reply:
Also applies to all the below logs, maybe have a helper function to output str + pk of a model

Reply:
Agree on printing user info (email address since that's unique), but don't understand why we need to print anything except Section pk. Other attributes can mutate, but the pk never will.

Reply:
My point is that if other attributes mutate, this log is no longer useful, since we care about the state of the system at the time of the log event